vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -66,34 +63,97 @@ def _apply_filters(
     return data_frame
 
 
-def _apply_filter_interaction(
-    data_frame: pd.DataFrame,
-    ctds_filter_interaction: List[CallbackTriggerDict],
-    target: str,
+def _apply_chart_filter_interaction(
+    data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
 ) -> pd.DataFrame:
-    for ctd in ctds_filter_interaction:
-        if not ctd["value"]:
+    ctd_click_data = ctd_filter_interaction["clickData"]
+    if not ctd_click_data["value"]:
+        return data_frame
+
+    source_chart_id: ModelID = ctd_click_data["id"]
+    source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    try:
+        custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
+    except KeyError as exc:
+        raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
+
+    customdata = ctd_click_data["value"]["points"][0]["customdata"]
+
+    for action in source_chart_actions:
+        if target not in action.function["targets"]:
             continue
+        for custom_data_idx, column in enumerate(custom_data_columns):
+            data_frame = data_frame[data_frame[column].isin([customdata[custom_data_idx]])]
 
-        source_chart_id = ctd["id"]
-        source_chart_actions = _get_component_actions(model_manager[source_chart_id])
+    return data_frame
 
-        try:
-            custom_data_columns = model_manager[source_chart_id]["custom_data"]  # type: ignore[index]
-        except KeyError as exc:
-            raise KeyError(f"No `custom_data` argument found for source chart with id {source_chart_id}.") from exc
 
-        customdata = ctd["value"]["points"][0]["customdata"]
+def _get_parent_vizro_table(_underlying_table_id: str) -> Table:
+    from vizro.models import Table
 
-        for action in source_chart_actions:
-            if target not in action.function["targets"]:
-                continue
+    for _, table in model_manager._items_with_type(Table):
+        if table._underlying_table_id == _underlying_table_id:
+            return table
+    raise KeyError(f"No parent Vizro.Table component found for underlying table with id {_underlying_table_id}.")
+
+
+def _apply_table_filter_interaction(
+    data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
+) -> pd.DataFrame:
+    ctd_active_cell = ctd_filter_interaction["active_cell"]
+    ctd_derived_viewport_data = ctd_filter_interaction["derived_viewport_data"]
+    if not ctd_active_cell["value"] or not ctd_derived_viewport_data["value"]:
```

Comment:
Could you remind me again, are `active_cell` and `derived_viewport_data` the only possible arguments? Or is it working like in the dcc.Graph and there are a few more? What other arguments exist?

Reply:
If the source of the filter interaction is `Graph` then `"clickData"` is the callback input.
If the source of the filter interaction is `Table` then `"active_cell"` and `"derived_viewport_data"` are the callback inputs.

Reply:
I meant in general e.g. in the Graph model I know the default callback input is `clickData` but in the future we also want to enable `hoverData`, `selectedData`, etc.

Do several potential options also exist for the data table? Or is it just going to be `active_cell `and `derived_viewport_data`?

Reply:
Ah, I see, sorry.
There are always multiple potential properies for any dash object so there is for a dash datatable too. 
e.g. If we want to enable something like the old model interactions, we should enable recognition of which rows in datatable are selected, so we should also support the `"derived_virtual_selected_rows"` property.